Installation.md

```diff
@@ -0,0 +1,259 @@
+
+```
+$ createuser tmc -s -P
+```
+
+If you cannot run the above command, you can also create the user manually
+
+```
+$ sudo -u postgres psql -c "CREATE USER tmc WITH SUPERUSER CREATEUSER CREATEDB PASSWORD 'tmc';"
+```
+:exclamation: Superuser access is useful for dev environment, but discouraged for production.
+
+#### Set PostgreSQL to authenticate with md5 instead of local ####
+
+Locate the **pg_hba.conf** file with `locate pg_hba.conf`. The file should be located */etc/postgresql/<version number>/main/pg_hba.conf*. Open the file with your preferred text-editor. You need root privileges to edit the file. Around line 90 change
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     peer
+```
+to
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     md5
+```
+
+### TMC-server installation ###
+#### Clone the TMC repository ####
+
+```
+$ cd tmc-server
+$ git clone https://github.com/testmycode/tmc-server.git
+```
+:exclamation: If you are not using a github account, replace the repository submodule URLs with a HTTPS URL in .git/config e.g. `https://github.com/testmycode/tmc-checkstyle-runner.git`
+```
+$ git submodule update --init --recursive
+$ gem install bundler && bundle install
+```
+You can view the site settings from the file `config/site.defaults.yml`. If you want to change the settings for the site, create a new file `config/site.yml` and define the changes there (notice: you do not need to copy the entire file. Settings not in `site.yml` will be looked up from `site.defaults.yml`).
+
+Initialize the database with `rake db:reset`
+#### Build sandbox ####
+
+```
+$ cd ext/tmc-sandbox
+$ sudo make
+```
+
+#### Install tmc-sandbox web dependencies ####
+
+```
+$ cd web
+$ bundle install
+$ rake ext
+$ rvmsudo rake test
+```
+
+#### Compile rest of the externals ####
+
+```
+$ cd ../..
+$ rake compile
+```
+
+#### Install tmc-check ####
+
+```
+$ sudo apt-get install check
+$ cd tmc-sandbox/uml/output/tmc-check
+$ rvmsudo make rubygems install clean
+```
+
+#### Run the test suite ####
+In the tmc-server root directory run
+```
+$ rvmsudo rake spec
+```
+
+### Post-install instructions ###
+
+#### Development setup ####
+Use `script/dev_env` to start the server in [screen](http://www.gnu.org/software/screen/). Please read through the file, if you are interested in the procedure behind starting the service.
+
+In screen, press `Ctrl+A` and then a number key to switch between tabs. To stop the dev environment, press `Ctrl+C`, wait a bit, then press `Q`. Repeat until all tabs are closed.
+
+The default user account is `admin`/`admin`.
+
+### Production setup ###
+
+To enable pghero stats (optional), add the following lines to postgresql.conf:
```

Comment:
Pghero really nice for prod but also might be nice for dev... Should we move this outside of Prod setup.